src/socket/address.cr

```diff
@@ -21,6 +21,27 @@ class Socket
       end
     end
 
+    # :ditto:
+    def self.from(sockaddr : LibC::Sockaddr*) : Address
+      case family = Family.new(sockaddr.value.sa_family)
+      when Family::INET6
+        sockaddr = sockaddr.as(LibC::SockaddrIn6*)
+
+        IPAddress.new(sockaddr, sizeof(typeof(sockaddr)))
+      when Family::INET
+        sockaddr = sockaddr.as(LibC::SockaddrIn*)
+
+        IPAddress.new(sockaddr, sizeof(typeof(sockaddr)))
+      when Family::UNIX
+        sockaddr = sockaddr.as(LibC::SockaddrUn*)
+
+        IPAddress.new(sockaddr, sizeof(typeof(sockaddr)))
```

Comment:
A `LibC::SockaddrUn` must construct a `Socket::UNIXAddress`, not `Socket::IPAddress`

Reply:
@HertzDevil oops... fixed